third_party/xla/xla/service/gpu/runtime/gpublas_lt_matmul_thunk.cc

```diff
@@ -35,8 +36,44 @@ limitations under the License.
 namespace xla {
 namespace gpu {
 
+struct MatmulPlanCache {
+
+  static MatmulPlanCache& i(const se::Stream *stream) {
+    static absl::Mutex m(absl::kConstInit);
+    // Each GPU gets different cache instance
+    static std::vector< std::unique_ptr< MatmulPlanCache > > meta(8);
+    absl::MutexLock lock(&m);
+    size_t dev_id = stream->parent()->device_ordinal();
+    if (dev_id >= meta.size()) meta.resize(dev_id + 1);
```

Comment:
ditto

Reply:
yes, that's correct.. it resizes the container so that meta[dev_id] becomes a valid element

Reply:
so shall we try to reduce the number of resize() and make it `meta.resize(dev_id + 8);` instead of 1?